sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/EventHubConsumerClient.cs

```diff
@@ -840,13 +841,20 @@ void exceptionCallback(Exception ex)
                         // Receive items in batches and then write them to the subscribed channels.  The channels will naturally
                         // block if they reach their maximum queue size, so there is no need to throttle publishing.
 
-                        receivedItems = await transportConsumer.ReceiveAsync(BackgroundPublishReceiveBatchSize, BackgroundPublishingWaitTime, cancellationToken).ConfigureAwait(false);
+                        if (receivedItems == default)
+                        {
+                            receivedItems = await transportConsumer.ReceiveAsync(BackgroundPublishReceiveBatchSize, BackgroundPublishingWaitTime, cancellationToken).ConfigureAwait(false);
+                            receivedItems = (receivedItems as IList<EventData>) ?? receivedItems.ToList();
```

Comment:
The AMQP transport consumer emits a batch of events as a list; the eager cast prevents making a copy of existing enumerable if it can be directly cast and falls back to allocating only if necessary.